### PR TITLE
backlight breathing overhaul

### PR DESCRIPTION
**Pull request by BalzGuenat**
This is a refactor of the breathing code that also fixes https://github.com/qmk/qmk_firmware/issues/2181.

### CODE REVIEW

**Comment by BalzGuenat** [comment]
Currently having trouble with the meira board by @colemarkham. He seems to have a pretty unusual setup for backlight and breathing. For example, in rules.mk, he has `BACKLIGHT_ENABLE = no` but then uses it as if enabled. Also, he defines `BACKLIGHT_BREATHING` in the `meira.c` file, which is not where you'd find that typically.

I hope he can look over this and either bring his files more in line with other keyboards or tell me how his implementation is supposed to work.

File: docs/faq_keymap.md [hunk 1]
```diff
@@ -127,7 +127,7 @@ https://github.com/tekezo/Karabiner/issues/403
 
 ## Esc and <code>&#96;</code> on a Single Key
 
-See the [Grave Escape](feature_grave_escape.md) feature.
+See the [Grave Escape](feature_grave_esc.md) feature.
```
Comment (BalzGuenat) [inline docs/faq_keymap.md]: This has nothing to do with the rest but I put in the fix anyway,

**Comment by BalzGuenat** [comment]
These things should also fix the issues described here: https://github.com/qmk/qmk_firmware/issues/1415

**Comment by jackhumbert** [comment]
This is awesome :) thanks!

**Pull request merged**